src/backend/distributed/commands/index.c

```diff
@@ -589,6 +589,13 @@ PreprocessReindexStmt(Node *node, const char *reindexCommand,
 
 		if (isCitusRelation)
 		{
+			if (PartitionedTable(relationId))
```

Comment:
What about `PartitionTable` s? Do they suffer from the same issue? You can create indexes with `ONLY` keyword to create indexes only on the partitions

Reply:
Added a test with a partition table. They work fine.
Also REINDEX doesn't seem to have ONLY.